packages/office-ui-fabric-react/src/utilities/hooks/useResponsiveMode.tsx

```diff
@@ -0,0 +1,27 @@
+import * as React from 'react';
+import { getWindow } from '@uifabric/utilities';
+import { useOnEvent } from '@uifabric/react-hooks';
+import { ResponsiveMode, getResponsiveMode, getInitialResponsiveMode } from '../decorators/withResponsiveMode';
+
+export const useResponsiveMode = (elementRef: React.RefObject<HTMLElement | null>) => {
+  const [lastResponsiveMode, setLastResponsiveMode] = React.useState<ResponsiveMode>(getInitialResponsiveMode());
```

Comment:
```suggestion
  const [lastResponsiveMode, setLastResponsiveMode] = React.useState<ResponsiveMode>(getInitialResponsiveMode);
```

If you pass in the function without calling it, you can avoid calling it on every render. `useState` default can accept a function which will only be called on initialization.